lib/swaps/Swaps.ts

```diff
@@ -926,6 +926,10 @@ class Swaps extends EventEmitter {
     } else if (this.getPendingSwapHashes().includes(rHash)) {
       throw errors.PAYMENT_PENDING(rHash);
     } else {
+      const dealInstance = await this.repository.getSwapDeal(rHash);
+      if (dealInstance && dealInstance.rPreimage && (dealInstance.state === SwapState.Completed || dealInstance.state === SwapState.Recovered)) {
```

Comment:
I think it's probably safer (and removes some unnecessary checks) to return the preimage as long as we have it, regardless of the swapstate. I think one of the objectives is to return the preimage to as many requests as possible to ensure we can settle any inbound payments, even if there's some data inconsistency in the database otherwise.